charmdir.go

```diff
@@ -425,9 +424,16 @@ func (dir *CharmDir) MaybeGenerateVersionString() (string, string, error) {
 		cmdArgs = []string{"bzr", "version-info"}
 	} else if _, err = os.Stat(filepath.Join(dir.Path, ".git")); err == nil {
 		cmdArgs = []string{"git", "describe", "--dirty", "--always"}
+	} else if file, err := os.Open(dir.join("version")); err == nil {
+		logger.Debugf("charm is not in version control and uses a version file")
+		var versionNumber = ""
+		_, err = fmt.Fscan(file, &versionNumber)
```

Comment:
I wonder if we use `fmt.Fscanf` instead, so that we only grab the number and nothing more?

```sh
var versionNumber string
n, err = fmt.Fscan(file, "%s", &versionNumber)
if err != nil {
	return "", "", err
}
if n != 1 {
	return "", "", errors.Errorf("unexpected a version")
}
```

Reply:
Also handling the `err` from `fmt.Fscan`

Reply:
thanks for the err catch!
I would like to stay consistent with the others how they handle file parsing. 
I actually want to make sure that the version file is correct and otherwise we log so (?)